### PR TITLE
add badges plugin

### PR DESCRIPTION
**Pull request by tardyp**
This ports the badge functionnality from eight to nine.
We just take the badges from eight and put them back to the source code.

after writting that, I found https://github.com/thefinn93/BuildbotStatusShields/tree/nine
 from @thefinn93 probably we should take this rendering engine instead of the hardcoded pngs.

## Contributor Checklist:

* [ ] I have updated the unit tests
* [x] I have created a file in the `master/buildbot/newsfragment` directory (and read the `README.txt` in that directory)
* [x] I have updated the appropriate documentation

### CODE REVIEW

**Comment by tardyp** [comment]
@thefinn93 would you like to port your renderer on top of this?

File: master/docs/developer/www-base-app.rst [hunk 1]
```diff
@@ -83,6 +83,10 @@ The entrypoint containing a Resource, nothing forbids plugin writers to add more
 For that, a reference to the master singleton is provided in ``master`` attribute of the Application entrypoint.
 You are even not restricted to twisted, and could even `load a wsgi application using flask, django, etc <http://twistedmatrix.com/documents/13.1.0/web/howto/web-in-60/wsgi.html>`_.
 
+It is also possible to make a web plugin which only adds http endpoint, and has no javascript UI.
+For that the ``Application`` endpoint object should have ``ui=False`` argument.
+You can look at the :src:`www/badges` plugin for an example of a ui-less plugin.
```
Comment (Frodox) [inline master/docs/developer/www-base-app.rst]: a**n**/the ui-less, I think

**Comment by thefinn93** [comment]
@tardyp I'd love to, but don't have the time. If you'd like to take the code and integrate it, feel free.

**Comment by tardyp** [comment]
@thefinn93 I've ported your renderer. looks much better

**Pull request merged**